### ISSUE TITLE
Corrupted FairRoot prefix after calling cmake twice

### ISSUE THREAD

**Issue by TobiasStockmanns** [FairRootGroup/FairRoot#1203]
**Describe the bug**
If cmake is called twice in PandaRoot either explicity via `cmake -S ./source -B ./build` or implicitly via make the paths to the `generated_dicitionary_root.sh.in` file gets corrupted.
The error message is:
```
CMake Error: File /Users/tstockmanns/PandaSoftware/FairRoot/dev/install/lib/cmake/FairRoot-19.0.0/share/fairbase/cmake/scripts/generate_dictionary_root.sh.in does not exist.
CMake Error at /Users/tstockmanns/PandaSoftware/FairRoot/dev/install/share/fairbase/cmake/modules/ROOTMacros.cmake:107 (Configure_File):
  Configure_File Problem configuring file
  ```
The correct path would be `/Users/tstockmanns/PandaSoftware/FairRoot/dev/install/share/fairbase/cmake/scripts/generate_dictionary_root.sh.in`.

Looking at the output of the cmake call the variable: `FairRoot prefix` changes between the first and second call:
- `--   FairRoot prefix            : /Users/tstockmanns/PandaSoftware/FairRoot/dev/install`
- `--   FairRoot prefix            : /Users/tstockmanns/PandaSoftware/FairRoot/dev/install/lib/cmake/FairRoot-19.0.0`

Most probably this is not a problem of PandaRoot but caused by the cmake scripts coming with FairRoot.

**Comment by ChristianTackeGSI** [comment]
Thanks for the report!

I am putting some notes here, what I suspect. I will try to look deeper in the next days.

I guess, the problem is in FindFairRoot.cmake:
https://github.com/FairRootGroup/FairRoot/blob/5fab55df73fb3a51bdd81160279b08548851fc4b/cmake/modules/FindFairRoot.cmake#L11-L12

* In the first run, `FairRoot_DIR` is not set. All works normally.
* In the first run a config mode `find_package` finally happens:
  https://github.com/FairRootGroup/FairRoot/blob/5fab55df73fb3a51bdd81160279b08548851fc4b/cmake/modules/FindFairRoot.cmake#L68
  This sets `FairRoot_DIR` as a cache variable to the path containing the `FairRootConfig.cmake` config mode package (or I would assume so).
* The cache variable persits…
* During the second run, the above `if(FairRoot_DIR` code is executed and redirects `FAIRROOTPATH` to the wrong path
* Things go wrong

The quick and dirty solution is probably to disable the `if` by replacing it with something like `if(FALSE)`.

**Comment by ChristianTackeGSI** [comment]
Will be fixed in 18.8 and 19.
